### ISSUE TITLE
[Feature]: remove X-Powered-By header 

### ISSUE THREAD

**Issue by sdavids** [remix-run/remix#1710]
### What is the new or updated feature that you are suggesting?

The `X-Powered-By` header should not be set

### Why should this feature be included?

> [...] then at least disable the X-Powered-By header. Attackers can use this header (which is enabled by default) to detect apps running Express and then launch specifically-targeted attacks.

http://expressjs.com/en/advanced/best-practice-security.html#at-a-minimum-disable-x-powered-by-header

**Comment by franklinjavier** [comment]
You can set by yourself 


```tsx
// entry.server.tsx
  responseHeaders.set("X-Powered-By", "Hugs");
```

![image](https://user-images.githubusercontent.com/440220/151886493-3d07892f-76f6-4c15-b2d2-4b831539322a.png)

**Comment by nakleiderer** [comment]
Unfortunately, using `responseHeaders.set("X-Powered-By", "Hugs")` does not remove the Express header. It concats with the express header producing `Express, Hugs` as the value

![image](https://user-images.githubusercontent.com/4278631/152607819-8262bba3-f079-4627-9e6a-7dad2a023135.png)

This can easily be fixed by adding the following line to Remix's server:
```js
app.disable('x-powered-by');
```

**Comment by sdavids** [comment]
#1712 has been merged ...

I am unsure how issues get resolved @ryanflorence 

Do I close this issue now? Or when are issues closed? By whom? Label "Release X"?

**Comment by mjackson** [comment]
Fixed in #1712
